### PR TITLE
docs: add usage about htc mock

### PR DESCRIPTION
**Pull request by esoubiran-aneo**

### CODE REVIEW

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Then, you can update the tag version. You can verify the [latest of Core on GitHub](https://github.com/aneoconsulting/ArmoniK.Core/release/latest) in order to use it.
+
+:warning: `dockerhubaneo/armonik_control` `dockerhubaneo/armonik_pollingagent` and `dockerhubaneo/armonik_core_htcmock_test_worker` must have the **exact** same tag version. It's a necessity to be sure that they can talk each other using the same API.
+
+Then, you can deploy ArmoniK as usual.
+
+```bash [shell]
+make deploy-armonik
+```
```
Comment (melflitty) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:37]: You need here to deploy all the components using splitted deployment not just armonik

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+```bash [shell]
+docker run --rm \
+            -e HtcMock__NTasks=100 \
+            -e HtcMock__TotalCalculationTime=00:00:00.100 \
+            -e HtcMock__DataSize=1 \
+            -e HtcMock__MemorySize=1 \
+            -e HtcMock__EnableFastCompute=true \
+            -e HtcMock__SubTasksLevels=1 \
+            -e HtcMock__Partition=htcmock \
```
Comment (melflitty) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:49]: Concerning the partition, if you do what you described in this doc: changing the worker to be htcmock, you will have an htcmock worker but in the default partition. In that case you don't need to specify the partition. (and specifying it to htcmock will not work I think)

Comment (melflitty) [reply .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:49]: You have different possibilities : 

1. make the deployment all in one : in that case a partition htcmock with worker htcmock is deployed and you don't need to change anything in the code but think about changing the partition to htcmock when doing the test
2. make the splitted deployment: either you do what you described in this doc changing the worker to htcmock in default partition OR add a partition htcmock with worker htcmock. @YKharouni do you confirm?

Comment (melflitty) [reply .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:49]: I am not sure if we need to  present all these possibilities in the doc

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# How to use HTC Mock?
+
+HTC Mock is a test tool for ArmoniK. It is used to generate tasks that will be runned by ArmoniK. It provides a lot of options to personalize the tasks.
+
+HTC Mock is used in CI to test ArmoniK automatically but it can also be used in development to generate realistic tasks.
+
+::alert{type="info"}
+To populate the database, you can also use some scripts to populate database. See [Populate database?](../populate-database.md)
```
Comment (pmilaneo) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:8]: To populate the database, you can also use scripts.  See [Populate database](../populate-database.md)

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    worker = [
+      {
+        name              = "worker"
+-       image             = "dockerhubaneo/armonik_worker_dll"
++       image             = "dockerhubaneo/armonik_core_htcmock_test_worker"
+        tag               = "0.12.1"
+    ]
+```
+
+Then, you can update the tag version. You can verify the [latest of Core on GitHub](https://github.com/aneoconsulting/ArmoniK.Core/release/latest) in order to use it.
```
Comment (pmilaneo) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:29]: Then, you can update the tag version. Please verify the [latest of Core on GitHub](https://github.com/aneoconsulting/ArmoniK.Core/release/latest) in order to use it.

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+:warning: `dockerhubaneo/armonik_control` `dockerhubaneo/armonik_pollingagent` and `dockerhubaneo/armonik_core_htcmock_test_worker` must have the **exact** same tag version. It's a necessity to be sure that they can talk each other using the same API.
```
Comment (pmilaneo) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:31]: :warning: `dockerhubaneo/armonik_control` `dockerhubaneo/armonik_pollingagent` and `dockerhubaneo/armonik_core_htcmock_test_worker` must have the **exact** same tag version. It is necessary so they can talk to each other using the same API.

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md (hunk 2 above)
Comment (esoubiran-aneo) [reply .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:49]: About the "all-in-one" deployment, there is no documentation about it so we can't talk about it for the moment.

About the name of the partition, maybe, we need to use the default one. Need to try these cases.

**Review by pmilaneo — CHANGES_REQUESTED** [review]
just a few changes and it is ok for me

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md (hunk 4 above)
Comment (pmilaneo) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:29]:
```suggestion
Then, you can update the tag version. Please verify the [latest of Core on GitHub](https://github.com/aneoconsulting/ArmoniK.Core/release/latest) in order to use it.
```

**Comment by aneojgurhem** [comment]
@esoubiran-aneo @pmilaneo Can one of you finish the last modifications so that we can integrate it ?

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md (hunk 1 above)
Comment (aneojgurhem) [reply .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:37]: I think we can refer to the related doc instead

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md (hunk 2 above)
Comment (aneojgurhem) [reply .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:49]: You should mention it

**Review by aneojgurhem — COMMENTED** [review]
@pmilaneo can you add the last bit of information to resolve the last comments ?

File: .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
++       image             = "dockerhubaneo/armonik_core_htcmock_test_worker"
+        tag               = "0.12.1"
+    ]
+```
+
+Then, you can update the tag version. Please verify the [latest of Core on GitHub](https://github.com/aneoconsulting/ArmoniK.Core/release/latest) in order to use it.
+
+:warning: `dockerhubaneo/armonik_control` `dockerhubaneo/armonik_pollingagent` and `dockerhubaneo/armonik_core_htcmock_test_worker` must have the **exact** same tag version. It is necessary so they can talk to each other using the same API.
+
+Then, you can deploy ArmoniK as usual.
```
Comment (pmilaneo) [inline .docs/content/2.guide/1.how-to/how-to-use-htc-mock.md:33]:
```suggestion
Then, you can deploy ArmoniK as explained in the [All in one deployment guide](https://aneoconsulting.github.io/ArmoniK/installation/linux/all-in-one-deployment).
```

**Review by pmilaneo — DISMISSED** [review]
added link to the all in on deployment guide

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**